### ISSUE TITLE
LateInitializationError: Field 'screenshotManager' has not been initialized.

### ISSUE THREAD

**Issue by heitor598** [ThexXTURBOXx/catcher_2#10]
## Describe the bug
LateInitializationError: Field 'screenshotManager' has not been initialized.

## To Reproduce
1. build the application
2. See error

## Expected behavior
That the `screenshotManager` variable is started correctly

## Screenshots
![Captura de Tela 2023-10-23 às 20 04 42](https://github.com/ThexXTURBOXx/catcher_2/assets/62099813/3e011532-f6d5-46ba-b662-a84cce90717b)

## Device (please complete the following information!)
 - Device: Xiaomi Resmi note 11
 - OS: macOs ventura 13.5.1
 - `catcher_2` version: [e.g. 1.0.0]

**Comment by ThexXTURBOXx** [comment]
This looks like a very early crash in your initialisation code before the screenshot manager was able to set itself up.
Maybe, there is something I can do to fix this, but I doubt that is the case. I will take a look, though.

**Comment by ThexXTURBOXx** [comment]
Are you also maybe able to share the function that you are passing to `runAppFunction`?
Maybe, the problem could also be avoided by passing your main app widget as argument to `rootWidget` instead of providing a `runAppFunction`

**Comment by heitor598** [comment]
I used the `rootWidget` argument but got the same problem.

**Comment by heitor598** [comment]
```dart
void main() async {
  . . .
    Catcher2(
      runAppFunction: () => runApp(const AppWidget()),
      debugConfig: settings.debugOptions,
      releaseConfig: settings.releaseOptions,
      navigatorKey: appNavigatorKey,
    );
}
```

**Comment by ThexXTURBOXx** [comment]
Okay, then it seems to be something else... Which Flutter version are you using?

**Comment by heitor598** [comment]
```
Flutter 3.13.8 • channel stable 
Tools • Dart 3.1.4 • DevTools 2.25.0
```

**Comment by ThexXTURBOXx** [comment]
Thanks for the information, I have tried reproducing the bug, but I can't.
Does this also happen with the example of this package for you?
Also, here is one of my apps which uses Catcher 2 and works perfectly fine: https://github.com/ThexXTURBOXx/studipassau

**Comment by ThexXTURBOXx** [comment]
@narumi147 Usually I would agree, but in this case, `late` is used to get rid of many occurrences of null-safe operators. Also, now the initialisation order of the variables is correct and this error should not happen anymore - I hope I can also say with a high amount of certainty that nothing can even go wrong in this regard anymore, but let's see...

**Comment by ThexXTURBOXx** [comment]
The fix should now be available as part of version `1.0.1`

**Comment by ThexXTURBOXx** [comment]
@narumi147 Could you please open a separate issue about this then? A stack trace from a failing Windows device would be very nice to appropriately address that issue.

**Comment by ThexXTURBOXx** [comment]
I would like to know which kind of error is raised and whether there is a way to fix it either at its core or using different parameters to get the device info.
Also, by that logic I should not trust anything, also getting the package info etc.
It would be a bit short-minded to just catch all the errors, I think

**Comment by ThexXTURBOXx** [comment]
@narumi147 Fix is now available as part of `1.0.2`

**Comment by ThexXTURBOXx** [comment]
New fix available in version `1.0.3`

**Comment by ThexXTURBOXx** [comment]
I have tested the new way of retrieving device info using
```dart
  Future<AndroidDeviceInfo> debugStuff() async =>
      rootWidget == null ? throw Error() : AndroidDeviceInfo.fromMap({});
```
and forcing to throw the error by specifying below:
```patch
-final androidInfo = await deviceInfo.androidInfo;
+final androidInfo = await debugStuff;
```
This forces the error to be thrown.
Since in my tests only a warning is logged and catching errors still works as usual, everything is solved here in my eyes.
Also, since this is a completely different problem than `LateInitializationError: Field 'screenshotManager' has not been initialized.` I think it is perfectly fine to say that the crash you reported should be filed as a separate issue.
The two have nothing to do with each other.

As for the `late` specifiers: I think I made my point above: They make sense in this case, as long as the initialisation order is correct - which it is now.
So the cause of this problem is fixed, which is why I closed it.
